M2/Macaulay2/packages/HomologicalShiftIdeals.m2

```diff
@@ -0,0 +1,1001 @@
+-- -*- coding: utf-8 -*-
+newPackage(
+        "HomologicalShiftIdeals",
+        Version => "1.0", 
+        Date => "May 9, 2022",
+        Authors => {{Name => "Antonino Ficarra", Email => "antficarra@unime.it"}},
+        Headline => "A Macaulay2 package for computing the homological shift ideals of a monomial ideal of a polynomial ring",
+        DebuggingMode => false,
+        PackageImports => {"SimplicialComplexes", "SimplicialDecomposability"},
+        Reload => true
+        )
+
+export {"supportIdeal",
+        "isFullySupported",
+        "toMonomial",
+        "toMultidegree",
+        "boundingMultidegree",
+        "multigradedShifts",
+        "HS",
+        "socle",
+        "hasLinearResolution",
+        "hasHomologicalLinearResolution",
+        "hasLinearQuotients",
+        "hasHomologicalLinearQuotients",
+        "admissibleOrder",
+        "isAdmissibleOrder",
+        "isPolymatroidal",
+        "isHomologicalPolymatroidal"
+};
+
+mydoc := "";
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    HomologicalShiftIdeals
+  Headline
+    a package for computing the homological shift ideals of a monomial ideal of a polynomial ring
+  Description
+    Text
+      HomologicalShiftIdeals is a package for computing the homological shift ideals of a monomial ideal of a polynomial ring
+///);
+
+-------------------------------------------------------------------------------------------
+-- Compute the support of a monomial ideal
+-------------------------------------------------------------------------------------------
+mydoc = concatenate(mydoc,///
+Node
+  Key
+    supportIdeal
+    (supportIdeal,Ideal)
+  Headline
+    compute the support of a monomial ideal
+  Usage
+    supportIdeal(I)
+  Inputs
+    I:Ideal
+      a monomial ideal $I$ of a polynomial ring $S = K[x_1,\ldots,x_n]$, $K$ a field
+  Outputs
+    l:List
+      the support of $I$
+  Description
+    Text
+      The support of a monomial ideal $I$ of a polynomial ring $S=K[x_1,\ldots,x_n]$ is the set $$\text{supp}(I)=\bigcup_{u\in G(I)}\text{supp}(u),$$ where $G(I)$ is the minimal monomial generating set of $I$ and $\text{supp}(u)=\{x_j:x_j\ \text{divides}\ u\}$ for a monomial $u$.
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2,x_3,x_1^2*x_4);
+      supportIdeal(I)
+    Example
+      S = QQ[x_1..x_4];
+      I = ideal(x_1^2,x_2^5,x_3,x_1*x_4);
+      supportIdeal(I)
+///);
+
+supportIdeal = method(TypicalValue=>List)
+supportIdeal (Ideal) := I -> (
+        if isMonomialIdeal(I)==false then return "expected a monomial ideal"
```

Comment:
There's no reason to compare a boolean to `true` or `false` inside an if statement like this.  `if not isMonomialIdeal I then ...` accomplishes the same thing.

Reply:
Also, you probably want `error` instead of `return` here.